OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -132,14 +132,39 @@ void IBotTick.BotTick(IBot bot)
 
 		bool ShouldBuildMCV()
 		{
+			// Only build MCV if we have a factory to build it with.
+			var enoughFactories = AIUtils.CountBuildingByCommonName(Info.McvFactoryTypes, player) > 0;
+			if (!enoughFactories)
+				return false;
+
 			// Only build MCV if we don't already have one in the field.
-			var allowedToBuildMCV = AIUtils.CountActorByCommonName(Info.McvTypes, player) == 0;
-			if (!allowedToBuildMCV)
+			var mcvCount = AIUtils.CountActorByCommonName(Info.McvTypes, player);
+			if (mcvCount > 0)
+				return false;
+
+			// Only build MCV if the conyardCount + mcvCount are lower than the minimum conyard count.
+			var conyardCount = AIUtils.CountBuildingByCommonName(Info.ConstructionYardTypes, player);
+			if (conyardCount + mcvCount >= Info.MinimumConstructionYardCount)
```

Comment:
`mcvCount` can only ever be zero here (since a greater number would be filtered out above), so this can simplify to
`if (conyardCount >= Info.MinimumConstructionYardCount)`? (And update the comment.)

Reply:
Good spot.